src/microbe_stage/editor/CellEditorComponent.cs

```diff
@@ -865,6 +867,19 @@ public override bool CanFinishEditing(IEnumerable<EditorUserOverride> userOverri
             return false;
         }
 
+        // This is triggered when no changes have been made
+        if (Editor.MutationPoints == Constants.BASE_MUTATION_POINTS)
+        {
+            TutorialState tutorialState = Editor.CurrentGame.TutorialState;
+
+            if (tutorialState.Enabled && !tutorialState.MadeNoChangesTutorial.HasBeenShown)
+            {
+                tutorialState.SendEvent(TutorialEventType.MicrobeEditorNoChangesMade, EventArgs.Empty, this);
+
+                return false;
```

Comment:
Instead of what the if is on line 875 this could be futureproofed a bit by changing that if to:
`if (tutorialState.Enabled) `

and then this return to be like this:

```
// Disallow exiting if a tutorial wanted to open
if(tutorialState.TutorialActive())
   return false
```

That way this code is no longer tightly coupled to the made no changes tutorial, but any tutorial can respond to the no changes made message and prevent exiting the editor.

Reply:
That check is there to prevent the "return false" from triggering after the tutorial has already been shown.

Reply:
Oh wait sorry you're right, I misread your comment